test/environment_v2/test_problems.py

```diff
@@ -0,0 +1,115 @@
+    assert isinstance(problem.start_time, datetime)
+    assert isinstance(problem.end_time, datetime)
+    assert all(isinstance(ae, EntityStub) for ae in problem.affected_entities)
+    assert all(isinstance(ae.entity_id, EntityId) for ae in problem.affected_entities)
+    assert all(isinstance(ie, EntityStub) for ie in problem.impacted_entities)
+    assert all(isinstance(ie.entity_id, EntityId) for ie in problem.impacted_entities)
+    assert isinstance(problem.root_cause_entity, EntityStub)
+    assert all(isinstance(mz, ManagementZone) for mz in problem.management_zones)
+    assert all(isinstance(et, METag) for et in problem.entity_tags)
+    assert all(isinstance(pf, AlertingProfileStub) for pf in problem.problem_filters)
+    assert isinstance(problem.evidence_details, pb.EvidenceDetails)
+    assert all(isinstance(e, pb.Evidence) for e in problem.evidence_details.details)
+    assert all(isinstance(e.entity, EntityStub) for e in problem.evidence_details.details)
+    assert all(isinstance(e.grouping_entity, EntityStub) for e in problem.evidence_details.details)
+    assert isinstance(problem.recent_comments, pb.CommentList)
+    assert all(isinstance(c, pb.Comment) for c in problem.recent_comments.comments)
+    assert isinstance(problem.impact_analysis, pb.ImpactAnalysis)
+    assert all(isinstance(i, pb.Impact) for i in problem.impact_analysis.impacts)
+
+    # value checks
+    assert problem.problem_id == PROBLEM_ID
+    assert problem.display_id == "P-210617"
+    assert problem.title == "Multiple infrastructure problems"
+    assert problem.impact_level == pb.ImpactLevel.INFRASTRUCTURE
+    assert problem.severity_level == pb.SeverityLevel.RESOURCE_CONTENTION
+    assert problem.status == pb.Status.CLOSED
+    assert all(ae.entity_id.id == "PROCESS_GROUP_INSTANCE-8092E71D6FBB914E" for ae in problem.affected_entities)
+    assert all(ae.entity_id.type == "PROCESS_GROUP_INSTANCE" for ae in problem.affected_entities)
+    assert all(ae.name == "easytravel.customer.frontend" for ae in problem.affected_entities)
+    assert all(ie.entity_id.id == "PROCESS_GROUP_INSTANCE-8092E71D6FBB914E" for ie in problem.impacted_entities)
+    assert all(ie.entity_id.type == "PROCESS_GROUP_INSTANCE" for ie in problem.impacted_entities)
+    assert all(ie.name == "easytravel.customer.frontend" for ie in problem.impacted_entities)
+    assert problem.root_cause_entity.entity_id.id == "PROCESS_GROUP-C44FB250621B8036"
+    assert problem.root_cause_entity.entity_id.type == "PROCESS_GROUP"
+    assert problem.root_cause_entity.name == "easytravel.customer.frontend"
+    assert problem.management_zones[0].id == "8692695975020499402"
+    assert problem.management_zones[0].name == "Operations Team"
+    assert all(et.context == "CONTEXTLESS" for et in problem.entity_tags)
+    assert problem.start_time == int64_to_datetime(1622807640000)
+    assert problem.end_time == int64_to_datetime(1622807820000)
+    assert problem.evidence_details.total_count == 4
```

Comment:
Add a 

`assert len(problem.evidence_details.details) == 4` 

This will fail, I believe the `EvidenceDetail` parsing is not working